test/unit_test/components/servo_test.dart

```diff
@@ -0,0 +1,229 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:grpc/grpc.dart';
+import 'package:viam_sdk/src/components/servo/client.dart';
+import 'package:viam_sdk/src/components/servo/service.dart';
+import 'package:viam_sdk/src/components/servo/servo.dart';
+import 'package:viam_sdk/src/gen/component/servo/v1/servo.pbgrpc.dart';
+import 'package:viam_sdk/src/resource/manager.dart';
+
+class FakeServo extends Fake implements Servo {
+  int angle = 0;
+  bool isStopped = true;
+  Map<String, dynamic>? extra;
+
+  final String name;
+
+  FakeServo(this.name);
+
+  @override
+  Future<void> move(int angle, {Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    this.angle = angle;
+    isStopped = false;
+  }
+
+  @override
+  Future<int> getPosition({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return angle;
+  }
+
+  @override
+  Future<bool> isMoving() async {
+    return !isStopped;
+  }
+
+  @override
+  Future<void> stop({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    isStopped = true;
+  }
+}
+
+void main() {
+  group('FakeServo Tests', () {
+    late FakeServo servo;
+    late String name;
+    final int testPosition = 42;
+    setUp(() {
+      name = 'servo';
+      servo = FakeServo(name);
+    });
+    test('angle should start at 0', () {
+      expect(servo.angle, 0);
+    });
+
+    test('isStopped should start at true', () {
+      expect(servo.isStopped, true);
+    });
+
+    test('name should be constructed with the value passed in', () {
+      servo = FakeServo('test');
+      expect(servo.name, 'test');
+    });
+
+    test('move should move angle to new position', () async {
+      await servo.move(testPosition);
+      expect(servo.angle, testPosition);
+    });
+
+    test('move should set isStopped to false', () async {
+      await servo.move(testPosition);
+      expect(servo.isStopped, false);
+    });
+
+    test('stop should set isStopped to true', () async {
+      servo.isStopped = false;
+      await servo.stop();
+      expect(servo.isStopped, true);
+    });
+
+    test('getPosition should return the angle', () async {
+      int testResult = await servo.getPosition();
+      expect(testResult, servo.angle);
+    });
+
+    test('doCommand', () async {
+      // TODO implement servo doCommand
+      expect(true, true);
+    });
+  });
+
+  group('Servo Service Tests', () {
+    late ClientChannel channel;
+    late FakeServo servo;
+    late ServoService service;
+    late Server server;
+    late String name;
+    final int testPosition = 42;
+
+    setUp(() async {
```

Comment:
do `setUp` and `tearDown` get called before and after each `test` respectively?

Reply:
They do. setUpAll() / tearDownAll() exist also if i wanted to run just once. I debated that option, but opted for a fresh start each time to limit unintended effects of testing on the same service / FakeServo.